mitmproxy/net/http/encoding.py

```diff
@@ -10,18 +10,22 @@
 import zlib
 import brotli
 
-from typing import Union
+from typing import Union, Optional  # noqa
 
 
 # We have a shared single-element cache for encoding and decoding.
 # This is quite useful in practice, e.g.
 # flow.request.content = flow.request.content.replace(b"foo", b"bar")
 # does not require an .encode() call if content does not contain b"foo"
-CachedDecode = collections.namedtuple("CachedDecode", "encoded encoding errors decoded")
+CachedDecode = collections.namedtuple(
+    "CachedDecode", "encoded encoding errors decoded"
+)
 _cache = CachedDecode(None, None, None, None)
 
 
-def decode(encoded: Union[str, bytes], encoding: str, errors: str='strict') -> Union[str, bytes]:
+def decode(
+    encoded: Optional[bytes], encoding: str, errors: str='strict'
+) -> Optional[str]:
```

Comment:
This should be `-> Optional[AnyStr]`. For example `gzip` decodes from gzipped to ungzipped bytes. See https://github.com/mitmproxy/mitmproxy/blob/master/test/mitmproxy/net/http/test_encoding.py